dll/win32/kernel32/client/file/dir.c

```diff
@@ -908,9 +908,9 @@ RemoveDirectoryW(IN LPCWSTR lpPathName)
     }
 
     /* Prepare to delete mount point */
-    RtlInitUnicodeString(&PathName, lpPathName);
-    PathName.Buffer = RtlAllocateHeap(RtlGetProcessHeap(), 0, PathName.Length + 2 * sizeof(WCHAR));
-    if (!PathName.Buffer)
+    PathName = RtlAllocateHeap(RtlGetProcessHeap(), 0, wcslen(lpPathName) * sizeof(WCHAR) +
```

Comment:
Note that the advantage of using the `UNICODE_STRING`, was that it wasn't needed to invoke twice the wcslen(lpPathName) : here and below (line 923) (otherwise you would need a separate variable to hold the string length, which is a bit what the `UNICODE_STRING` structure offers), and, that in a certain sense, it was "ensured" (by truncation at least...) that the path would be no longer than MAXUSHORT.